diff/content_diff.go

```diff
@@ -49,28 +49,94 @@ func getContentDiff(config *Config, state *state, content1, content2 openapi3.Co
 
 func getContentDiffInternal(config *Config, state *state, content1, content2 openapi3.Content) (*ContentDiff, error) {
 
-	result := newContentDiff()
+	initialResult := newContentDiff()
+	processedDeleted := make(map[string]bool) // Keep track of deleted items that found an equivalent added item
 
+	// 1. Find exact matches and initial deletions
 	for name1, media1 := range content1 {
-		if media2, ok := content2[name1]; ok {
-			diff, err := getMediaTypeDiff(config, state, media1, media2)
+
+		if media2, ok := content2[name1]; ok { // Exact match found
+
+			diff, err := getMediaTypeDiff(config, state, name1, name1, media1, media2)
 			if err != nil {
 				return nil, err
 			}
 
 			if !diff.Empty() {
-				result.MediaTypeModified[name1] = diff
+				initialResult.MediaTypeModified[name1] = diff
 			}
+			// Mark as processed (implicitly, by not adding to deleted)
 		} else {
-			result.MediaTypeDeleted = append(result.MediaTypeDeleted, name1)
+			// No exact match, potential deletion
+			initialResult.MediaTypeDeleted = append(initialResult.MediaTypeDeleted, name1)
 		}
 	}
 
+	// 2. Find initial additions
 	for name2 := range content2 {
 		if _, ok := content1[name2]; !ok {
-			result.MediaTypeAdded = append(result.MediaTypeAdded, name2)
+			initialResult.MediaTypeAdded = append(initialResult.MediaTypeAdded, name2)
 		}
 	}
 
-	return result, nil
+	// 3. Iteratively find equivalent pairs from initial Added/Deleted lists
+	finalAdded := make(utils.StringList, 0, len(initialResult.MediaTypeAdded))
+	finalModified := initialResult.MediaTypeModified // Start with exact matches
+
+	for _, addedName := range initialResult.MediaTypeAdded {
+		foundEquivalentDeleted := false
```

Comment:
If the count of added and deleted is both 1, we could also consider that as a change, even if the content also changed 🤔

Reply:
I understand the idea but I think it could lead to errors.
For example, if media-type changed from application/xml to application/json.